ete3/coretype/tree.py

```diff
@@ -2076,44 +2454,73 @@ def convert_to_ultrametric(self, tree_length=None, strategy='balanced'):
         """
 
         # Could something like this replace the old algorithm?
-        #most_distant_leaf, tree_length = self.get_farthest_leaf()
-        #for leaf in self:
+        # most_distant_leaf, tree_length = self.get_farthest_leaf()
+        # for leaf in self:
         #    d = leaf.get_distance(self)
         #    leaf.dist += (tree_length - d)
-        #return
-
+        # return
 
+        # get origin distance to root
+        dist2root = {self: 0.0}
+        for node in self.iter_descendants("levelorder"):
+            dist2root[node] = dist2root[node.up] + node.dist
 
-        # pre-calculate how many splits remain under each node
-        node2max_depth = {}
-        for node in self.traverse("postorder"):
-            if not node.is_leaf():
-                max_depth = max([node2max_depth[c] for c in node.children]) + 1
-                node2max_depth[node] = max_depth
-            else:
-                node2max_depth[node] = 1
-        node2dist = {self: 0.0}
+        # get tree length by the maximum
         if not tree_length:
-            most_distant_leaf, tree_length = self.get_farthest_leaf()
+            tree_length = max(dist2root.values())
         else:
             tree_length = float(tree_length)
 
+        if strategy == "weighted":
+            # get remaining distance to tree tips
+            dist2tip = {self: tree_length}
+            # modify the dist property of nodes
+            for node in self.iter_descendants("levelorder"):
+                if node.is_leaf():
+                    node.dist = dist2tip[node.up]
+                else:
+                    # weight the distance of children nodes by their median
+                    cdist = (
+                        statistics.median(dist2root[l] for l in node) - dist2root[node]
```

Comment:
optional? add statistics.mean support?

Reply:
use `utils.mean` which is already imported